website/docs/basics/autowiring.md

```diff
@@ -0,0 +1,218 @@
+---
+id: autowiring
+title: Autowiring
+sidebar_label: Autowiring
+---
+
+[![docs-source](https://img.shields.io/badge/source-eigthshift--libs-blue?style=for-the-badge&logo=php&labelColor=2a2a2a)](https://github.com/infinum/eightshift-libs)
+
+What is autowiring?
+
+*Autowiring is an exotic word that represents something very simple: the container's ability to create and inject dependencies automatically.*
+
+The quote was used from [PHP-DI](https://php-di.org/doc/autowiring.html) library that we use.
+
+In a nutshell, we use a dependency injection container to load all your service classes. With DI container, we also use the PSR-4 coding standard that allows us to predict how your file/folder structure will look like in correlation to the namespace. With that in mind, you don't have to worry about when your class is required, and when it loads, our autowiring takes care of everything. You need to follow a few coding standards.
+
+Autowiring runs inside the `Main` service class as an extended class in our `AbstractMain` class. You can install the main class using this command:
+
+`wp boilerplate create_main`
+
+## How it works?
+
+Here is a quick overview of how this works:
+
+- You create a new class manually or using the WP-CLI command.
+- PSR-4 would detect your new class if you followed the PSR-4 naming standard.
+- New class is added to the `classmap array` inside the `vendor` folder.
+- Autowiring class reads the new class and checks if you have any DI classes used in your new class.
+- It just works.
+
+As you saw from the overview steps, autowiring checks your class and loads DI classes accordingly.
+
+### What if my class doesn't have anything inside a constructor method?
+
+If your class **doesn't have** anything defined in the constructor method, autowiring will require your class.
+This just means that you are not using DI.
+
+### What if my class does have a primitive parameter inside a constructor method?
+
+If your class ** has** a primitive parameter defined in the constructor method, autowiring will check what type it is and load it to your new class.
```

Comment:
If you have a primitive you need to define it at the point of usage. If that point of usage is services array, then yes, you'll need to define them there.

I mean, since we are trying to have just one point of origin for our dependencies (the service array) it's kinda valid in all cases I think 🤔

Reply:
👍